app/src/main/java/com/github/polypoly/app/utils/UniqueCodeGenerator.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.polypoly.app.utils
+
+import android.annotation.SuppressLint
+import android.provider.Settings.System.getString
+import com.github.polypoly.app.R
+import com.github.polypoly.app.base.menu.lobby.GameLobby
+import com.github.polypoly.app.network.getValue
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_CODE_LENGTH
+import com.github.polypoly.app.utils.Constants.Companion.GAME_LOBBY_MAX_CHARACTERS
+import com.github.polypoly.app.utils.global.GlobalInstances
+import com.github.polypoly.app.utils.global.Settings
+import java.util.concurrent.CompletableFuture
+import kotlin.random.Random
+
+/**
+ * Generates a unique code for a game lobby
+ */
+class UniqueCodeGenerator(private val codeLength: Int = GAME_LOBBY_CODE_LENGTH) {
```

Comment:
What does the lobby has to do here? I believe you either should rename the class to `UniqueLobbyCodeGenerator` then, however this logic is global and could be very useful if we want to create other unique codes, it is up to you to choose :)

Reply:
I don't see where we would need to generate any other code than for lobby codes. We could imagine making this class modular to generate any type of code, but I don't think it is a necessary work yet. But you are right about the name not being precise, i will change it! :)